src/platform/nrfconnect/wifi/NrfWiFiDriver.h

```diff
@@ -54,10 +57,14 @@ class NrfWiFiDriver final : public WiFiDriver
 
     struct WiFiNetwork
     {
-        char ssid[DeviceLayer::Internal::kMaxWiFiSSIDLength];
-        uint8_t ssidLen = 0;
-        char credentials[DeviceLayer::Internal::kMaxWiFiKeyLength];
-        uint8_t credentialsLen = 0;
+        uint8_t ssid[DeviceLayer::Internal::kMaxWiFiSSIDLength];
+        size_t ssidLen = 0;
+        uint8_t pass[DeviceLayer::Internal::kMaxWiFiKeyLength];
+        size_t passLen = 0;
+
+        bool IsConfigured() const { return ssidLen > 0; }
+        ByteSpan GetSsid() const { return ByteSpan(ssid, ssidLen); }
```

Comment:
nit: I would consider renaming these methods to indicate that they return the `ByteSpan` instead of a raw pointer (like GetSsidByteSpan). Someone may confuse them with *direct* getters.

Reply:
I'm personally not a big fan of java-style long names causing a lot of line breaks as long as the type system protects my back from misusing the API, but I guess it's one of those subjective things, and I'm not totally against it, so I'll change it :)

Reply:
Yeh, it quite subjective. It's also not a strong opinion from my side. It's your call :)